### PR TITLE
feat(chat): implement marketplace UI (Epic #1988)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

implement Marketplace UI

Issues:

- Issue #1988 

**UI changes**

https://www.figma.com/design/lwL5jneveZczAj5RLcLFH2/DIAL-Chat?node-id=11630-38064&node-type=canvas&t=jQXx17KQRqIPqWIb-0

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by irinakartun — DISMISSED** [review]

**Comment by VolhaBazhkova** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Something went wrong, check [GitHub Workflow run output](https://github.com/epam/ai-dial-ci/actions/runs/11033506719)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: PR Workflow
 
 on:
   pull_request:
-    branches: [development, release-*]
+    branches: [development, release-*, feat/marketplace]
```
Comment (IlyaBondar) [inline .github/workflows/pr.yml]: revert it

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ModelVersionSelect.tsx [hunk 2]
```diff
@@ -37,6 +44,20 @@ export const ModelVersionSelect = ({
...
 
   if (entities.length < 2) {
+    if (entities.length && entities[0].version) {
+      return (
+        <p className="flex gap-2" data-qa="version">
+          {showVersionPrefix && (
+            <>
+              <span className="hidden md:block">{t('version: ')}</span>
+              <span className="md:hidden">{t('v: ')}</span>
+            </>
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ModelVersionSelect.tsx]: this block is used at least twice. Maybe better to create small internal component and use in both places and export it if somewhere we have the same rendering

File: apps/chat/src/utils/app/publications.ts [hunk 3]
```diff
@@ -22,9 +22,6 @@ import { getFolderIdFromEntityId, splitEntityId } from './folders';
 import { isRootId } from './id';
 import { EnumMapper } from './mappers';
 
-export const isItemPublic = (id: string) =>
-  id.split('/')[1] === PUBLIC_URL_PREFIX;
-
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/publications.ts]: Why did you remove good util-method istead of modifying it and use this logic directly in all places? DRY: don't repeat yourself and keep repeatable logic in one place
```
export const isItemPublic = (id: string, featureType: FeatureType) =>
  id.startsWith(
            getRootId({
              featureType,
              bucket: PUBLIC_URL_PREFIX,
            }));
```
but first version works without specifying of `featureType` and I like it more

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — DISMISSED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2102.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Review by irinakartun — APPROVED** [review]
